initrd/etc/functions

```diff
@@ -33,6 +33,28 @@ DO_WITH_DEBUG() {
 	"$@"
 }
 
+# Trace the current script and function.
+TRACE_FUNC() {
+	# Index [1] for BASH_SOURCE and FUNCNAME give us the caller location.
+	# FUNCNAME is 'main' if called from a script outside any function.
+	# BASH_LINENO is offset by 1, it provides the line that the
+	# corresponding FUNCNAME was _called from_, so BASH_LINENO[0] is the
+	# location of the caller.
+	TRACE "${BASH_SOURCE[1]}(${BASH_LINENO[0]}): ${FUNCNAME[1]}"
```

Comment:
Thank you so much for this! I forgot about this but that was goal when switching to bash!

Reply:
I cannot say enough how I'm happy to see this!